core/src/stored/backends/crud_storage.cc

```diff
@@ -0,0 +1,323 @@
+#include "include/bareos.h"
+#include <fmt/format.h>
+#include "crud_storage.h"
+#include "lib/berrno.h"
+#include "lib/bpipe.h"
+#include "lib/bstringlist.h"
+#include "stored/stored_conf.h"
+#include "stored/stored_globals.h"
+#include <sys/stat.h>
+#include <cctype>
+
+namespace {
+class BPipeHandle {
```

Comment:
Dont forget the rule of 5.  Currently this class allows copying (by copying the pointer value), which is not what we want/expect.

Maybe we can wrap `std::unique_ptr<BPipe, BPipeDestructor>` instead ?  In that case we would get the rule of 5 for free.

Reply:
Good catch! I disabled copy/move now. If we ever need to move such an object, we can easily add it.
I think the current abstraction is already leaky - a `unique_ptr` is probably worse.